src/OpenTracing/LogUtils.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace OpenTracing\LogUtils;
+
+use Exception;
+use InvalidArgumentException;
+use OpenTracing\LogField;
+use Throwable;
+
+function interleavedKVToFieldsConverter(array $keyValueFields)
+{
+    $fields = [];
+
+    foreach ($keyValueFields as $key => $value) {
+        if (!is_string($key)) {
+            throw new InvalidArgumentException(sprintf("Non-string key, got %s of type %s", $key, gettype($key)));
+        }
+
+        switch (true) {
+            case is_bool($value):
```

Comment:
@felixfbecker I am willing to change it but I struggled when looking for references about the performance here. Do you have anyone?